### PR TITLE
Feat render notion page

### PR DESCRIPTION
**Pull request by pacoyang**

### CODE REVIEW

File: src/components/notion-render/Image.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+const Image = ({ theAtom }: { theAtom: BlockAtom }) => {
+  const block = useAtomValue(theAtom) as ImageBlockObjectResponse
+  const imageUrl: string =
+    block.image.type == 'external'
+      ? block.image.external.url
+      : block.image.file.url
+  return (
+    <div className="notion_image_container">
+      <figure>
+        <img src={imageUrl} alt="" />
```
Comment (Leechael) [inline src/components/notion-render/Image.tsx:16]: `alt={block.image.caption}` for seo.

File: src/components/notion-render/Table.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import { useAtomValue } from 'jotai'
+import { TableBlockObjectResponse } from '@notionhq/client/build/src/api-endpoints'
+
+import { BlockAtom } from './atoms'
+import RichText from './RichText'
+
+const Table = ({ theAtom }: { theAtom: BlockAtom }) => {
+  const block = useAtomValue(theAtom) as TableBlockObjectResponse
```
Comment (Leechael) [inline src/components/notion-render/Table.tsx:8]: Open question: how to avoiding use `as` here?

File: src/components/notion-render/Todo.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+const Todo = ({
+  theAtom,
+  children,
+}: {
+  theAtom: BlockAtom
+  children: React.ReactNode
+}) => {
+  const block = useAtomValue(theAtom) as ToDoBlockObjectResponse
+  return (
+    <div className={`notion_${block.type}_container`}>
```
Comment (Leechael) [inline src/components/notion-render/Todo.tsx:16]: the `type` should always `to_do` here?

File: src/components/notion-render/atoms.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        ...theBlock,
+        numbered_list_item: {
+          ...theBlock.numbered_list_item,
+          listNumber,
+        },
+      }
+    }
+    return theBlock
+  })
+}
```
Comment (Leechael) [inline src/components/notion-render/atoms.ts:37]: Can we try `atomFamily` here?

File: src/lib/notion-client.ts [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+      })(property as never)
+      return null
+  }
+}
+
+export async function getParsedPageByProtertyText({
+  database_id,
+  property,
+  property_text,
+}: {
```
Comment (Leechael) [inline src/lib/notion-client.ts:241]: 1. typo: **Property**.

2. Can it redesign to:

```ts
getParsedPagesByProperties(database_id: string, properties: Record<string, string>): Promise<ParsedPage[]>;
```

File: src/components/notion-render/BulletedListItem.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import { useAtomValue } from 'jotai'
+import { BulletedListItemBlockObjectResponse } from '@notionhq/client/build/src/api-endpoints'
```
[the remark this answers is not in the document]
Comment (pacoyang) [reply src/components/notion-render/BulletedListItem.tsx:3]: Already done.

**Pull request merged**